### PR TITLE
Issue 370/add ignore to assigned tasks

### LINKED ISSUE #370: Add `ignored` state to assigned tasks
There is a new status for tasks called `ignored`, which is when a user has explicitly indicated that they will not complete the task.

Currently, this is not indicated in the UI any differently from any other incomplete task.

### PR DESCRIPTION
**Pull request by ziggabyte**
Resolves #370 

This PR adds a "ignored" status to tasks, an icon to tasks with that status, sorting to make ignored tasks appear between completed and not completed tasks, and tests to confirm that sorting works. 

In each sorted category the task with the most recent date will appear closest to the top.

### CODE REVIEW

File: src/components/ZetkinPerson.tsx
Comment (richardolsson) [inline src/components/ZetkinPerson.tsx:18]: What's the purpose of this change?

File: src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx
Comment (richardolsson) [inline src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx:9]: Am I right in thinking that this only tests one of the statuses, and does not confirm that the correct status is displayed when ignored or completed?

File: src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.tsx
Comment (richardolsson) [inline src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.tsx:32]: Could we display relative time here?

File: src/components/organize/tasks/TaskAssigneesList/index.tsx
Comment (richardolsson) [inline src/components/organize/tasks/TaskAssigneesList/index.tsx:25]: This could also be simplified (different logic, same result) as `return secondDate - firstDate;`

**Review by richardolsson — CHANGES_REQUESTED** [review]
Great work on this! I love to see all of these tests running and passing. 😊 

Would it be simple to display the date (relative) that the ignore happened? Similar to how it works for completed tasks. I think we should change this. What do you think?

image

I also left some minor comments below.

File: src/components/ZetkinPerson.tsx
Comment (ziggabyte) [reply src/components/ZetkinPerson.tsx:18]: this is bc of the "div can not be a child of p"-thing. If subtitle has the added flair of an icon, it will be inside a Box to make the text look good. and the Box (div) cannot be a child of the Typography (p).

File: src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx
Comment (ziggabyte) [reply src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx:9]: Yes, it only tests one, and only tests that it creates a notCompleted status. Should I add to test for the two others?

File: src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.tsx
Comment (ziggabyte) [reply src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.tsx:32]: Sounds like a good Idea!

File: src/components/organize/tasks/TaskAssigneesList/index.tsx
Comment (ziggabyte) [reply src/components/organize/tasks/TaskAssigneesList/index.tsx:25]: Cool!! thank you!

File: src/components/ZetkinPerson.tsx
Comment (richardolsson) [reply src/components/ZetkinPerson.tsx:18]: I see! Well, a `Box` (`<div>`) should probably also not be inside a `span`, even though there might not be anything complaining about it.

Would it not make sense to render this conditionally, based on whether `subtitle` is complex content (i.e. components) or just text?

What do you think @djbusstop? (No rush, answer tomorrow or Friday).

File: src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx
Comment (ziggabyte) [reply src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx:9]: @richardolsson should i add to the test you think?

File: src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx
Comment (richardolsson) [inline src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx:40]: The variable name `notCompletedSubtitle` here seems ambiguous. Is it a copy-paste mistake?

File: src/components/organize/tasks/TaskAssigneesList/index.tsx
Comment (richardolsson) [inline src/components/organize/tasks/TaskAssigneesList/index.tsx:20]: I'm not a fan of nested ternaries. Is there another way that this could be expressed that would be easier to read?

**Review by richardolsson — COMMENTED** [review]
More great work! I have some comments below that I would like to hear your thoughts on before I approve. 😊

File: src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx
Comment (ziggabyte) [reply src/components/organize/tasks/TaskAssigneesList/TaskStatusSubtitle.spec.tsx:40]: Definitely a copy-paste-mistake! fixing!

File: src/components/ZetkinPerson.tsx
Comment (richardolsson) [inline src/components/ZetkinPerson.tsx:17]: The line is very long, how could we break it up?

Also, do we need the `<Box>` here, or could it be up to the user of the component to create a box if necessary?

**Review by richardolsson — APPROVED** [review]
I think this looks good, so it's ready to merge if you want to.

But I did comment on one thing below, and I'll let you @ziggabyte decide whether to merge or make changes first. 😊

File: src/components/ZetkinPerson.tsx
Comment (richardolsson) [inline src/components/ZetkinPerson.tsx:20]: The fragment should not be needed here. I'm guessing you put it here so that you could use `{ subtitle }` (because the `{}` is only meaningful in JSX) but another way would have been to just remove `{}` and put just `subtitle` here.

**Review by richardolsson — DISMISSED** [review]
Looking good! Another comment below just FYI, but no need to make any changes. Go ahead and merge! 🎉

File: src/components/ZetkinPerson.tsx
Comment (djbusstop) [inline src/components/ZetkinPerson.tsx:18]: This needs the `variant="body2"` bit back in it. I will make a commit with this fix :)

**Review by djbusstop — CHANGES_REQUESTED** [review]

**Comment by djbusstop** [comment]
@richardolsson Made some small changes if you want to review. @ziggabyte great work on this :)

**Review by richardolsson — APPROVED** [review]
Looking good. Green light to merge! 🟢

**Pull request merged**
